config/network-policies.yaml

```diff
@@ -0,0 +1,566 @@
+      release: cf-db
+  ingress:
+  - from:
+    - podSelector:
+        matchLabels:
+          job-name: ccdb-migrate
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: cf-api-worker
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: cf-api-clock
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: cf-api-server
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: cf-api-deployment-updater
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: uaa
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+---
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: uaa
+  namespace: #@ data.values.system_namespace
+spec:
+  policyTypes:
+  - Ingress
+  - Egress
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/name: uaa
+  ingress:
+  - from:
+    - podSelector:
+        matchLabels:
+          istio: ingressgateway
+      namespaceSelector:
+        matchLabels:
+          cf-for-k8s.cloudfoundry.org/istio-system-ns: ""
+    - podSelector:
+        matchLabels:
+          app: log-cache
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: cf-api-server
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+  egress:
+  - {}
+---
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: cf-api-server
+  namespace: #@ data.values.system_namespace
+spec:
+  policyTypes:
+  - Ingress
+  - Egress
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/name: cf-api-server
+  ingress:
+  - from:
+    - podSelector:
+        matchLabels:
+          istio: ingressgateway
+      namespaceSelector:
+        matchLabels:
+          cf-for-k8s.cloudfoundry.org/istio-system-ns: ""
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: cf-api-kpack-watcher
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+    - podSelector:
+        matchLabels:
+          name: eirini
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+    - podSelector:
+        matchLabels:
+          app: log-cache
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+  egress:
```

Comment:
That's deny all, isn't it? This will collide with the [support for external databases](https://github.com/cloudfoundry/cf-for-k8s/issues/168). As the IPs of external databases might not be known or even fixed, I suppose we will have to allow egress traffic to all public IPs, potentially also Cloud Provider network internal IPs.

I would appreciate, if this doesn't get merged without a documented concept of how to open up the egress rules.
At the very least the [external database documentation](https://github.com/cloudfoundry/cf-for-k8s/blob/master/docs/configuration/external-databases.md) should be supported and updated.

Reply:
Ah ok, RTFM. 
```yaml
egress:
  - {}
```

is actually allow all. As this is work in progress, the general comment about keeping external databases in mind still stands.

Reply:
Yep, was keeping external databases, external service brokers, external identity providers, etc. in mind for Cloud Controller/UAA.